src/dcap_provider.cpp

```diff
@@ -898,19 +900,79 @@ static sgx_plat_error_t build_pck_crl_url(
     return SGX_PLAT_ERROR_OK;
 }
 
-static std::string build_tcb_info_url(const std::string& fmspc)
+char* base64_encode(const void* source)
+{
+    // Character set of base64 encoding scheme
+    char char_set[] = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789+/";
+    const char* input = static_cast<const char*>(source);
+    int len_str = strlen(input);
+    char* res_str = (char*)malloc(SIZE * sizeof(char));

```

Comment:
This is hardcoded to 1000 any reason for that? Can this be based on size of the input?

Reply:
std::string result instead of res_str?